distributed/cli/tests/test_dask_worker.py

```diff
@@ -682,3 +684,116 @@ def dask_setup(worker):
         await c.wait_for_workers(1)
         [foo] = (await c.run(lambda dask_worker: dask_worker.foo)).values()
         assert foo == "setup"
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("nanny", ["--nanny", "--no-nanny"])
+def test_timeout(nanny):
+    worker = subprocess.run(
+        ["dask-worker", "192.168.1.100:7777", nanny, "--death-timeout=1"],
+        text=True,
+        encoding="utf8",
+        capture_output=True,
+    )
+
+    assert "timed out starting worker" in worker.stderr.lower()
+    assert "end worker" in worker.stderr.lower()
+    assert worker.returncode == 1
+
+
+@pytest.mark.slow
+@pytest.mark.skipif(WINDOWS, reason="POSIX only")
+@pytest.mark.parametrize("nanny", ["--nanny", "--no-nanny"])
+@gen_cluster(client=True, nthreads=[])
+async def test_sigint(c, s, nanny):
+    try:
+        worker = subprocess.Popen(
+            ["dask-worker", s.address, nanny],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        await c.wait_for_workers(1)
+        worker.send_signal(signal.SIGINT)
+        stdout, stderr = worker.communicate()
+        logs = stdout.decode().lower()
+        assert stderr is None
+        if nanny == "--nanny":
+            assert "closing nanny" in logs
+        else:
+            assert "nanny" not in logs
+        assert "stopping worker" in logs
+        assert "end worker" in logs
+        assert "timed out" not in logs
+        assert "signal" not in logs
+        assert "error" not in logs
+        assert "exception" not in logs
+        assert worker.returncode == 0
+    finally:
+        worker.kill()
+
+
+@pytest.mark.slow
+@pytest.mark.skipif(WINDOWS, reason="POSIX only")
+@pytest.mark.parametrize("nanny", ["--nanny", "--no-nanny"])
+@gen_cluster(client=True, nthreads=[])
+async def test_sigterm(c, s, nanny):
+    try:
+        kwargs = {}
+        worker = subprocess.Popen(
+            ["dask-worker", s.address, nanny],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        await c.wait_for_workers(1)
+
+        worker.send_signal(signal.SIGTERM)
+        stdout, stderr = worker.communicate()
+        logs = stdout.decode().lower()
+        assert stderr is None
+        assert f"signal {signal.SIGTERM}" in logs
+        if nanny == "--nanny":
+            assert "closing nanny" in logs
+        else:
+            assert "nanny" not in logs
+        assert "stopping worker" in logs
+        assert "end worker" in logs
+        assert "timed out" not in logs
+        assert "error" not in logs
+        assert "exception" not in logs
+        assert worker.returncode == 0
+    finally:
+        worker.kill()
+
+
+@pytest.mark.slow
+@pytest.mark.skipif(not WINDOWS, reason="Windows only")
+@pytest.mark.parametrize("nanny", ["--nanny", "--no-nanny"])
+@gen_cluster(client=True, nthreads=[])
+async def test_ctrl_break_event(c, s, nanny):
```

Comment:
I'd drop this test for now since it looks like getting it to work properly will be more complicated. I also don't have a Windows machine to manually check whether the script behaves as it should, which adds to the problem.

Reply:
@sjperkins and I have tested manually that keyboard interrupts on the `dask-scheduler` and `dask-worker` produce the same results as on my Ubuntu machine.

Reply:
Dropped the test after offline discussion.